sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuMultiFileReader.scala

```diff
@@ -280,6 +336,110 @@ object MultiFileReaderUtils {
       partsCols
     }
   }
+
+  private def buildPartitionsColumns(partRowNumsAndValues: Array[(Long, InternalRow)],
+      partSchema: StructType): Array[GpuColumnVector] = {
+    // build the partitions vectors for all partitions within each column
+    // and concatenate those together then go to the next column
+    closeOnExcept(new Array[GpuColumnVector](partSchema.length)) { partsCols =>
+      for ((field, colIndex) <- partSchema.zipWithIndex) {
+        val dataType = field.dataType
+        withResource(new Array[ColumnVector](partRowNumsAndValues.length)) { onePartCols =>
+          partRowNumsAndValues.zipWithIndex.foreach { case ((rowNum, valueRow), partId) =>
+            val singleValue = valueRow.get(colIndex, dataType)
+            withResource(GpuScalar.from(singleValue, dataType)) { oneScalar =>
+              onePartCols(partId) = ColumnVector.fromScalar(oneScalar, rowNum.toInt)
+            }
+          }
+          val res = if(onePartCols.length > 1) {
+            ColumnVector.concatenate(onePartCols: _*)
+          } else {
+            onePartCols.head
+          }
+          partsCols(colIndex) = GpuColumnVector.from(res, field.dataType)
+        }
+      }
+      partsCols
+    }
+  }
+
+  /**
+   * Splits the (partRows, partValues) into multiple batches such that sum
+   * of sizes does not exceed cudf limit
+   *
+   * Example,
+   * Input:
+   *  partRows:   [ 10, 40, 70, 10, 11 ]
+   *  partValues: [ [abc, ab], [bc, ab], [abc, bc], [aa, cc], [ade, fd] ]
+   *
+   * Assume:
+   *  limit:      300 bytes
+   *
+   * Split Calculation:
+   *  At rowIndex = 2:
```

Comment:
I think you need to mention how the maxValues are calculated on a per entry basis... ie first set of rows column 0 maxVAlue=3 but second set of rows maxValue column0 = 2.

Reply:
Updated documentation